src/rts/GameState.java

```diff
@@ -886,4 +890,53 @@ public static GameState fromJSON(String JSON, UnitTypeTable utt) {
         return gs;
     }
 
+    /**
+     * Constructs a matrix ovservation for a player
+     * @param player
+     * @return a matrix observation for the specified player
+     */
+    public int [][][] getMatrixObservation(int player){
```

Comment:
similarly, mabye "JNIgetMatrixObservation"? (or something like that). Basically, let's mark the JNI functions in the API in some way, so it's clear what they are for.